build_cmake.bat

```diff
@@ -3,14 +3,44 @@ rmdir build /s /q
 
 mkdir build
 
+set DOCKER_IMAGE=forderud/qtwasm:latest
+
+:: Docker buildsetup
+set LOCAL_BUILD_FOLDER=%cd%\build
+set LOCAL_SOURCE_FOLDER=%cd%\sample
+set DOCKER_BUILD_FOLDER=/project/build
+set DOCKER_SOURCE_FOLDER=/project/source
+
 :: Input sources must be found in "sample" subfolder
 :: Output binaries will be written to "build" subfolder
 
-:: Default build settings
-docker run --rm -v %cd%\sample:/project/source -v %cd%\build:/project/build forderud/qtwasm:latest || exit /b 1
+:: Valid choices are "Debug", "RelWithDebInfo", "Release" (default)
+set BUILD_MODE=Release
+if not "%1"=="" set BUILD_MODE=%1
+
+:: Copy wasm.cmake for now (need the gsource-map)
+:: Simplify later when wasm.camke is part of the image (can skip the QT_TOOLCHAIN_FILE argument)
+copy image\wasm.cmake sample
+docker run --rm -v %LOCAL_SOURCE_FOLDER%:%DOCKER_SOURCE_FOLDER% -v %LOCAL_BUILD_FOLDER%:%DOCKER_BUILD_FOLDER% %DOCKER_IMAGE% /bin/bash -c "/opt/Qt/bin/qt-cmake -DQT_CHAINLOAD_TOOLCHAIN_FILE=/project/source/wasm.cmake -DCMAKE_BUILD_TYPE=%BUILD_MODE% -G Ninja /project/source && ninja" || exit /b 1
+
+if "%BUILD_MODE%"=="Release" (
+    run_webserver.bat
+) else (
+    :: Set up paths to match the relative paths during build step
+    :: and also keep all needed files within what is being served
+    set DOCKER_DEBUG_BUILD_FOLDER=/build
+    set DOCKER_DEBUG_SOURCE_FOLDER=/source
+    set LOCAL_WEBSERVER_PY=%cd%\WebServer.py
+    set DOCKER_DEBUG_SERVE_FOLDER=/
+    set DOCKER_WEBSERVER_PY=/WebServer.py
 
-:: Custom build settings
-::copy image\wasm.cmake sample
-::docker run --rm -v %cd%\sample:/project/source -v %cd%\build:/project/build forderud/qtwasm:latest /bin/bash -c "/opt/Qt/bin/qt-cmake -DQT_CHAINLOAD_TOOLCHAIN_FILE=/project/source/wasm.cmake -DCMAKE_BUILD_TYPE=Release -G Ninja /project/source && ninja" || exit /b 1
+    start http://localhost:8080/build/app/helloworld.html
 
-run_webserver.bat
+    :: Serve website from within docker to get consistent source mapping
+    docker run -p 8080:8080 ^
```

Comment:
please pass `--rm` as done above to avoid risk of old containers filling up the disk.